include/mitsuba/render/kdtree.h

```diff
@@ -1103,20 +1205,21 @@ template <typename BoundingBox_, typename Index_, typename CostModel_,
 
             Scalar left_cost = 0, right_cost = 0;
 
-            BuildTask &left_task = *new (allocate_child()) BuildTask(
+            BuildTask left_task = BuildTask(
                 m_ctx, children, std::move(partition.left_indices),
                 left_bounds, partition.left_bounds, m_depth+1,
                 m_bad_refines, &left_cost);
 
-            BuildTask &right_task = *new (allocate_child()) BuildTask(
-                m_ctx, std::next(children), std::move(partition.right_indices),
+            BuildTask right_task = BuildTask(
+                m_ctx, children + 1, std::move(partition.right_indices),
                 right_bounds, partition.right_bounds, m_depth + 1,
                 m_bad_refines, &right_cost);
 
-            set_ref_count(3);
-            spawn(left_task);
-            spawn(right_task);
-            wait_for_all();
+            Task *left_ek_task = ek::do_async([&](){ left_task.execute(); });
```

Comment:
There are two small ways of improving this code:

1. We don't need to call `do_async` for both tasks. Instead, we can execute one on the current thread (which is waiting) and push another one onto the job queue.
2. We should switch over to serial execution and completely avoid the `do_async` once the tasks become small enough. (GRAIN_SIZE sounds reasonable)